scripts/run-related-test.mjs

```diff
@@ -0,0 +1,67 @@
+/**
+ * This script finds all files under the paths that contain a marker and reads lines after the marker from the files.
+ * The marker is defined as a comment line that starts with `// TEST:`.
+ * The script is useful to find all related test cases for a given code change.
+ * Usage: `node scripts/run-related-test.mjs <path1> <path2> ...`
+ */
+
+import { promisify } from 'node:util'
+import { exec as execOrg } from 'node:child_process'
+
+const exec = promisify(execOrg)
+
+/**
+ * Find all files under the paths that contain a marker
+ * @param {string[]} paths - Paths to search in
+ * @param {string} marker - Marker to search for
+ */
+async function findFiles(paths, marker) {
+  const foundPaths = []
+  for (const path of paths) {
+    const command = `find "${path}" -type f -exec grep -l "${marker}" {} +`
```

Comment:
You can probably use `fs.readdir(path, { recursive: true }).filter(file => file.includes(marker)` instead

https://nodejs.org/api/fs.html#fspromisesreaddirpath-options